### ISSUE TITLE
[Feature Request] 支持通过 props 传入 `<Breadcrumb>` 配置

### ISSUE THREAD

**Issue by Zihan-Hu** [vuepress-theme-hope/vuepress-theme-hope#3822]
### Checklist

- [X] I searched existing issues and no one else requests similar feature.
- [X] I think that 25%+ users are positive towards this feature.

### Describe the feature

https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/054b3f5bbc6c8d6e7634392174c86bd1e10d39dc/packages/theme/src/client/components/BreadCrumb.ts#L63-L88

添加 prop `config`，传入 `config` 时不走这部分逻辑

### 起因

有时为了方便管理，文章会放在一些对读者没什么意义（或不会访问到）的目录里（比如 `/2024/01/01.md` `/05/4b3f5.md`）

### 最初的想法

像自定义文章类型那样？
然而 #3739 被否决力
<details>
<summary>展开瞎扯</summary>
<p>mdn 上对 breadcrumb 作用解释是这样的：</p>
<blockquote>
<p>A breadcrumb, or breadcrumb trail, is a navigational aid that is typically placed between a site's header and the main content, displaying either a hierarchy of the current page in relation to the site's structure, from top level to current page, or <strong><em>a list of the links the user followed to get to the current page, in the order visited</strong></em>.</p>
<p>...</p>
<p>Breadcrumb trails enable users to be aware of their location within a website. This type of navigation, if done correctly, <strong><em>helps users know where they are in a site and how they got there</strong></em>. They can also help a user get back to where they were before and can reduce the number of clicks needed to get to a higher-level page.</p>
</blockquote>
<p>如果 breadcrumb 里有用户不该知道、不该访问的目录，就不再是 <em>a list of the links the user followed to get to the current page</em>，失去了 <em>help users know how they got there</em> 的作用</p>
</details>

### 现在的想法

那就覆盖组件吧，好像只能跟随路由，那就加个手动传入的方法好了（idea from: `<RouterView>`）
或许不应该用类似 `/2024/01/01.md` 的目录结构，另一种方案是：
```
- article
  - 2023-07-20.md
  - 2023-07-21.md
  - 2023-07-22.md
  - ...
```
目前在用这个方案，每次都要花点时间翻文件列表，因为 `article` 目录还有其他命名格式不同的文件 :cry:

### 最后的办法

全文复制 `@theme-hope/components/BreadCrumb` 改改（悲
但是如果主题后面的版本改了 breadcrumb 组件，要手动同步 ~~“我上次改了什么来着？”~~

还是上游优先好了（逃

**Comment by Mister-Hope** [comment]
I personally refuse this change, BreadCrumb should be a feature out of box. Almost every user would not expect to set breadcrumb manually.

If you want to provide your own logic, component overriding is recommended. You are likely submitting a personal patch into our project, as definitely:

:x:  25%+ users are positive towards this feature.

**Comment by Zihan-Hu** [comment]
If someone decides to override the default BreadCrumb, it's required to rewrite a BreadCrumb component by himself.
However, more likely, just want to do special processing on certain paths, but not completely change the default behavior.

For a whopping 99.99% of users, this tweak can be overlooked because the default behavior of the theme won't change at all.

**Comment by Zihan-Hu** [comment]
**TL;DR**: More often, users only need to make some minor adjustments, rather than the entire replacement.

**Comment by Zihan-Hu** [comment]
Just like [slots](https://theme-hope.vuejs.press/guide/advanced/replace.html#using-slots) - If slots can be used, no one would like to rewrite a NormalPage component.

Perhaps only 0.01% of users will use it, but the other 99.99% would not object...

**Comment by Mister-Hope** [comment]
breadcrumb is not a complicated component.

If you think that a nice util.function could be extracted, then you can open a new pr to exact some.logic.out of.the component for you to reuse.
